pkg/controller/job/job_controller_test.go

```diff
@@ -274,7 +274,8 @@ func TestControllerSyncJob(t *testing.T) {
 		expectedPodPatches      int
 
 		// features
-		jobReadyPodsEnabled bool
+		jobReadyPodsEnabled   bool
+		podIndexLabelDisabled bool
```

Comment:
Reviewer note: Since the feature flag is default enabled, I inverted the usual parameter pattern `<featureflag>Enabled bool` flag to `<featureflag>Disabled bool`, so that all the existing test cases that don't specify this parameter will execute with the feature flag enabled, and then I added one test case which sets this parameter to true to execute indexed job test case with the feature flag disabled. This way we achieve full unit test coverage for a "default enabled" feature (which I think is important), while still having some verification that things work when it is disabled.

Reply:
Thanks for adding that!  I think it looks fine to me.  And thank you for the note.  It is a bit odd pattern but I can understand why you did it that one.